packages/shell-api/src/shell-instance-state.ts

```diff
@@ -74,7 +74,7 @@ export interface EvaluationListener extends Partial<ConfigProvider<ShellUserConf
   /**
    * Called when print() or printjson() is run from the shell.
    */
-  onPrint?: (value: ShellResult[]) => Promise<void> | void;
+  onPrint?: (value: ShellResult[], type: 'print' | 'printjson') => Promise<void> | void;
```

Comment:
Do we need/want to update browser-repl as well? I’m not sure what that would even look like.

Reply:
I am not sure that we would want to, yeah, doesn't look like this would make sense in browser-repl context to me too